x/auth/keeper/deterministic_test.go

```diff
@@ -59,10 +81,12 @@ func (suite *DeterministicTestSuite) SetupTest() {
 	queryHelper := baseapp.NewQueryServerTestHelper(suite.ctx, suite.encCfg.InterfaceRegistry)
 	types.RegisterQueryServer(queryHelper, suite.accountKeeper)
 	suite.queryClient = types.NewQueryClient(queryHelper)
+
+	suite.key = key
 }
 
 func (suite *DeterministicTestSuite) runAccountIterations(addr sdk.AccAddress, prevRes types.AccountI) {
```

Comment:
small nit: could you move this function below `createAndSetAccounts`? So that the Test* and run*Iterations functions are next to each other

Reply:
For this function I also think it's stronger to test the Anys directly:

```go
func (suite *DeterministicTestSuite) runAccountIterations(addr sdk.AccAddress, prevRes *codectypes.Any) {
	for i := 0; i < iterCount; i++ {
		acc, err := suite.queryClient.Account(suite.ctx, &types.QueryAccountRequest{Address: addr.String()})
		suite.Require().NoError(err)
		suite.Require().NotNil(acc)

		suite.Require().Equal(prevRes, acc.Account)
	}
}
```

This way, we don't only test the Any's values, but also how those values are packed in the Any